docker-storage-setup.sh

```diff
@@ -707,32 +707,56 @@ scan_disks() {
 }
 
 create_partition_sfdisk(){
-  local dev="$1" size
+  local dev="$1" size label="$2" type
   # Use a single partition of a whole device
   # TODO:
-  #   * Consider gpt, or unpartitioned volumes
+  #   * Consider unpartitioned volumes
   #   * Error handling when partition(s) already exist
   #   * Deal with loop/nbd device names. See growpart code
+
   size=$(( $( awk "\$4 ~ /"$( basename $dev )"/ { print \$3 }" /proc/partitions ) * 2 - 2048 ))
+  # if parted is not present, and this version of sfdisk
+  # does not support GPT then we are limited to 2TB
+  if [ $label = "gpt" ]; then
+    if $(sfdisk --help|grep gpt >/dev/null) ; then
+      label='label: gpt'
+      type=''
+      size=$(( $size - 34 ))
+    else
+      Info "Your version of sfdisk does not support GPT labels. Will use MBR instead but partition size will be limited to 2TB"
+      label=''
+      type=', Id=8e'
+      # 2TB in sectors - 2048
+      if [ $size -gt 4294965248 ]; then size=4294965248; fi
+    fi
+  elif [ $label = "msdos" ]; then
+    label=''
+  fi
+  
     cat <<EOF | sfdisk $dev
 unit: sectors
+$label
 
-${dev}1 : start=     2048, size=  ${size}, Id=8e
+${dev}1 : start=     2048, size=  ${size}${type}
 EOF
 }
 
 create_partition_parted(){
-  local dev="$1"
-  parted $dev --script mklabel msdos mkpart primary 0% 100% set 1 lvm on
+  local dev="$1" label="$2"
+  parted $dev --script mklabel $label mkpart primary 0% 100% set 1 lvm on
```

Comment:
If label is of type  `gpt` then keyword `primary` is not partition type any more. mkpart will expect a partition name. I am assuming right now it is using string `primary` as partition name instead.